src/LinuxTracing/LinuxTracingIntegrationTest.cpp

```diff
@@ -676,6 +676,14 @@ void VerifyCallstackSamplesWithOuterAndInnerFunction(
       // address and the caller address should match the "outer" function's address.
       if (callstack.pcs(pc_index) >= inner_function_virtual_address_range.first &&
           callstack.pcs(pc_index) <= inner_function_virtual_address_range.second) {
+        // Frame-pointer unwinding skips the second innermost frame if the sample fell on `push rbp`
+        // (which is one byte) or `mov rbp,rsp`. Disregard such samples. See b/179376436#comment4.
+        if (unwound_with_frame_pointers &&
+            (callstack.pcs(pc_index) == inner_function_virtual_address_range.first ||
+             callstack.pcs(pc_index) == inner_function_virtual_address_range.first + 1)) {
+          continue;
+        }
```

Comment:
should we fix unwinding instead?

Reply:
also it only makes sense for the top pc right?

Reply:
> should we fix unwinding instead?

The unwinding is done by perf_event_open.

> also it only makes sense for the top pc right?

You mean top of the callstack?

Reply:
> > also it only makes sense for the top pc right?

> You mean top of the callstack?

Added `pc_index == 0` to the `if`.